tests/common.py

```diff
@@ -53,6 +54,13 @@ def cleanUp(self):
         # Clear out thread local session cache
         CONN_CACHE.session = None
 
+    def get_session_factory(self, test_case):
+        if C7N_RECORD:
+            func = self.record_flight_data
```

Comment:
given the extant usage, i was thinking of just having replay do record if the env var is set... albeit that feels a bit more magical (replay does record vs get session dispatch) from a semantic naming perspective